test/index.js

```diff
@@ -808,4 +814,430 @@ describe('function helpers', () => {
       'Failed to download https://example.com/nonexistentfile: 404 Not Found',
     )
   })
+
+  describe('config', () => {
+    describe('generateCustomHeaders', () => {
```

Comment:
[question/potential boulder]: given this function will be used in the `onPostBuild` step of the plugin, is there any particular reason why a test (or an assertion within an existing test for `onPostBuild`) doesn't exist to confirm that `generateCustomHeaders` is called within that step? 

Doing so will ensure that the functionality is working in the area where we expect it to, and the more rigorous testing of the functionality can remain here.